modules/claims_api/app/services/claims_api/disability_compensation/docker_container_service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'pdf_generator_service/pdf_client'
+require 'claims_api/v2/disability_compensation_evss_mapper'
+require 'evss_service/base'
+
+module ClaimsApi
+  module DisabilityCompensation
+    class DockerContainerService < ClaimsApi::Service
+      def upload(claim_id)
+        log_service_progress(claim_id, 'docker_service',
+                             'Docker container service started')
+
+        auto_claim = get_claim(claim_id)
+        # Reset for a rerun on this
+        set_pending_state_on_claim(auto_claim) unless auto_claim.status == pending_state_value
+
+        evss_data = get_evss_data(auto_claim)
+
+        log_service_progress(claim_id, 'docker_service',
+                             'Submitting mapped data to Docker container')
+
+        evss_res = evss_service.submit(auto_claim, evss_data)
+
+        log_service_progress(claim_id, 'docker_service',
+                             "Successfully submitted to Docker container with response: #{evss_res}")
+        # update with the evss_id returned
+        auto_claim.update!(evss_id: evss_res[:claimId])
+        # clear out the evss_response value on successful submssion to docker container
+        clear_evss_response_for_claim(auto_claim)
+        # queue flashes service
+        queue_flash_updater(auto_claim.flashes, auto_claim&.id)
+        # now upload to benefits documents
+        start_bd_uploader_job(auto_claim) if auto_claim.status != errored_state_value
+      end
+
+      def get_evss_data(auto_claim)
+        evss_mapper_service(auto_claim, veteran_file_number(auto_claim)).map_claim
+      end
+    end
+  end
+end
```

Comment:
I am wondering what yours and the teams opinions on a slightly different approach here would be.  Since there is a decent amount of shared methods between the new sync service and the related async sidekiq jobs.  Wondering if a possible option would be a base class for each service that each could then inherit their shared methods from, and then from there the error handling and anything else unique to one style of run would then be in those individual files.  

My biggest concern here is that if we needed to change something in one of these jobs, like lets just say this line `set_pending_state_on_claim(auto_claim) unless auto_claim.status == pending_state_value` (for whatever reason).  We would need to remember to catch both now, which likely _could_ happen no problem, but I'm worried about all the duplication and maintainability down the road.  Curious about your thoughts on that, as well as the rest of the team's thoughts on that. 

_OR_ maybe it is worth thinking about, but as a next step, not sure.

Reply:
@rockwellwindsor-va yes, I have been thinking about this concern as well. Where are you suggesting this base class live?

Reply:
I do agree that we can probably find a lot of shared logic to inherit from a "base" class - unfortunately I think the pending check above is a bad example of that. In the synchronous flow, we have no option in either of these calls to have anything outside of PENDING. we don't toggle to anything besides ERRORED and any error should immediately terminate the flow and send back an error response to the consumer. If we don't have a PDF, we're dead in the water, so why try and establish a claim? To further that, the next request will create a NEW claim, so there's no way to get back to that status for this.

In respect to adding a base class for shared service features, I think that makes sense - the question is, once we "clean up" some of the stuff that's only relevant to the async flow, do we have enough of a need for this?

Reply:
@stiehlrod  I think Alex makes a really good point about if it is worth it or not after getting it dialed in.  Might be better left as a tech debt ticket to refactor if needed once we see.  Really my thought was that `modules/claims_api/app/services/claims_api/service.rb` could be the parent of the two base files since it seems like at least some of those methods are shared but  until the finished process is in place it is hard to tell if that is worth it or not.  Probably better left as tech debt, if needed at all.